src/automation/automation.ts

```
@@ -0,0 +1,254 @@
+
+    return automations;
+  } catch (error) {
+    console.error(`Error fetching automation list: ${JSON.stringify(error)}`);
+    return null;
+  }
+};
+
+/** ======================== List Automation Instances ======================== */
+
+export type Instance = {
+  ledgerId: string;
+  entityInfo: {
+    apiVersion: string;
+    artifactHash: string;
+    entity: {
+      tag: string;
+      value: {
+        tag: string;
+        value: AutomationValue;
+      };
+    };
+  };
+  enabled: boolean;
+  deployer: string;
+  config: {
+    tag: string;
+    value: {
+      name: string;
+      runAs: string;
+      configMap: {};
+    };
+  };
+  id: string;
+  instanceLabel: string | null;
+  createdAt: string;
+  owner: string;
+};
+
+const publishedInstance: Decoder<Instance> = object({
+  ledgerId: string(),
+  entityInfo: object({
+    apiVersion: string(),
+    artifactHash: string(),
+    entity: object({
+      tag: string(),
+      value: object({
+        tag: string(),
+        value: automationValue,
+      }),
+    }),
+  }),
+  enabled: boolean(),
+  deployer: string(),
+  config: object({
+    tag: string(),
+    value: object({
+      name: string(),
+      runAs: string(),
+      configMap: object(),
+    }),
+  }),
+  id: string(),
+  instanceLabel: oneOf(string(), constant(null)),
+  createdAt: string(),
+  owner: string(),
+});
+
+const listInstances: Decoder<Instance[]> = array(publishedInstance);
+export const listAutomationInstances = async (token: string): Promise<Instance[] | null> => {
+  const headers = {
+    Authorization: `Bearer ${token}`,
+    'Content-Type': 'application/json',
+  };
+
+  try {
+    const result = await fetch(`/.hub/v1/published/instance`, { headers, method: 'GET' });
+    const json = await result.json();
+    const instances = listInstances.runWithException(json);
+
+    return instances;
+  } catch (error) {
+    console.error(`Error fetching automation instances: ${JSON.stringify(error)}`);
```

Hmm…why don't we allow the error to continue propagating? A successful `null` in the case of a failure doesn't seem right here.

---

Scrolling down, I see that pattern a fair bit too…

---

Also is it worth considering retry logic here?

---

o I see... you think it'll be better to let the exceptions throw, and have the user of the library catch / deal with it? 

Retry logic should definitely be added in... maybe some generic API caller function (like we have in Hub's UI code) that does it for every call. I think I'll add this in a future PR with tests (`1.0.0-rc.2`)